src/main/webapp/sharing/container.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import gql from 'graphql-tag'
+import { useQuery } from '@apollo/react-hooks'
+import { useApolloFallback } from '../react-hooks'
+import Sharing from './sharing'
+import { useMutation } from '@apollo/react-hooks'
+
+const getSharingAttributes = id => {
+  const getMetacardSharing = gql`
+    query Sharing($id: ID!, $settings: QuerySettingsInput) {
+      metacardSharingAttributes(id: $id, settings: $settings) {
+        individuals {
+          read
+          write
+          admin
+        }
+        groups {
+          read
+          write
+        }
+      }
+      user {
+        roles
+      }
+    }
+  `
+  return useQuery(getMetacardSharing, { variables: { id } })
+}
+
+const fragment = gql`
+  fragment SharingAttributes on MetacardAttributes {
+    security_access_individuals_read
+    security_access_individuals
+    security_access_administrators
+  }
+`
+const saveMetacardSharing = () => {
+  const mutation = gql`
+    mutation saveMetacardSharing($id: ID!, $attrs: MetacardAttributesInput!) {
```

Comment:
kind of along the same lines as the last comment - if we remove the new resolver and use the generic metacard resolver, we can duplicate the same fragments for saving and getting sharing attributes

Reply:
My only concern with that is the client handling the data being reconstructed so that the sharing component can use it properly. But I'll look into just using the metacard resolver.